### PR TITLE
Add tests for the examples in the docs

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add tests for the files in "docs/examples" to ensure that changes in ORT
do not break the examples. The test class is added to the "cli" module,
because this module depends on all other modules and therefore has
access to all required classes.

### CODE REVIEW

File: docs/examples/go-dep.ort.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
-excludes:
+excludes: {}
```
Comment (sschuberth) [inline docs/examples/go-dep.ort.yml:1]: I guess these fixes were meant to go to a separate commit?

File: cli/src/funTest/kotlin/ExamplesTest.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.ossreviewtoolkit.reporter.reporters.AbstractNoticeReporter
+
+class ExamplesTest : StringSpec() {
+    private val examplesDir = File("../docs/examples")
+
+    init {
+        "ort.yml examples are parsable" {
+            examplesDir.listFiles { file ->
+                file.name.endsWith(".ort.yml")
+            }!!.forEach { file ->
```
Comment (sschuberth) [inline cli/src/funTest/kotlin/ExamplesTest.kt:48]: How about using `.orEmpty()` instead of `!!`?

Comment (mnonnenmacher) [reply cli/src/funTest/kotlin/ExamplesTest.kt:48]: I'd rather have the test fail if it cannot find the files.

File: docs/examples/go-dep.ort.yml (hunk 1 above)
Comment (mnonnenmacher) [reply docs/examples/go-dep.ort.yml:1]: No, they were required to make the new tests pass.

File: cli/src/funTest/kotlin/ExamplesTest.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/funTest/kotlin/ExamplesTest.kt:48]: Ok, just note that `listFile` returns `null` "if this abstract pathname does not denote a directory, or if an I/O error occurs", but not if there are no matching files.

File: docs/examples/go-dep.ort.yml (hunk 1 above)
Comment (sschuberth) [reply docs/examples/go-dep.ort.yml:1]: Yeah, I got that, but I still thought you might have wanted to separate fixing the examples from adding the tests. I'm fine with keeping the changes in one commit now that you've improved the commit message in that regard. However, (very) strictly speaking "were not valid YAML" is not correct for `docs/examples/sbt.ort.yml` which *was* valid YAML but contained an invalid ORT exclude reason.
